Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/CheckInRepository.kt

```diff
@@ -24,10 +26,23 @@ class CheckInRepository @Inject constructor(
         traceLocationDatabase.eventCheckInDao()
     }
 
+    /**
+     * Returns all stored check-ins
+     */
     val allCheckIns: Flow<List<CheckIn>> = checkInDao
         .allEntries()
         .map { list -> list.map { it.toCheckIn() } }
 
+    /**
+     * Returns check-ins that are within the retention period. Even though we have a worker that deletes all stale
+     * check-ins it's still possible to have stale check-ins in the database because the worker only runs once a day.
+     */
+    val checkInsWithinRetention: Flow<List<CheckIn>> = allCheckIns.map { checkInList ->
+        checkInList.filter { checkIn ->
+            isWithinRetention(checkIn, timeStamper)
```

Comment:
please pass a common instant , every time timeStamper.newUTC is called it is getting a new instance.
Maybe we should think of away to avoid this common mistake?

Reply:
Fixed. The only idea that I have to avoid this mistake would be to add some comments in the `TimeStamper`.